test/modules/EdgeHubRestartTester/src/Program.cs

```diff
@@ -0,0 +1,318 @@
+            {
+                iotHubServiceClient?.Dispose();
+                dmModuleClient?.Dispose();
+                msgModuleClient?.Dispose();
+            }
+
+            completed.Set();
+            handler.ForEach(h => GC.KeepAlive(h));
+            Logger.LogInformation("EdgeHubRestartTester Main() finished.");
+            return 0;
+        }
+
+        static TestResultBase CreateTestResult(
+            TestOperationResultType testOperationResultType,
+            DateTime edgeHubRestartedTime,
+            HttpStatusCode edgeHubRestartStatusCode,
+            DateTime completedTime,
+            HttpStatusCode completedStatus,
+            Guid batchId,
+            uint restartSequenceNumber,
+            long sequenceNumber)
+        {
+            switch (testOperationResultType)
+            {
+                case TestOperationResultType.Messages:
+                    return new EdgeHubRestartMessageResult(
+                        Settings.Current.ModuleId + testOperationResultType.ToString(),
+                        DateTime.UtcNow,
+                        Settings.Current.TrackingId,
+                        batchId.ToString(),
+                        sequenceNumber.ToString(),
+                        edgeHubRestartedTime,
+                        edgeHubRestartStatusCode,
+                        completedTime,
+                        completedStatus,
+                        restartSequenceNumber);
+
+                case TestOperationResultType.DirectMethod:
+                    return new EdgeHubRestartDirectMethodResult(
+                        Settings.Current.ModuleId + testOperationResultType.ToString(),
+                        DateTime.UtcNow,
+                        Settings.Current.TrackingId,
+                        batchId,
+                        sequenceNumber.ToString(),
+                        edgeHubRestartedTime,
+                        edgeHubRestartStatusCode,
+                        completedTime,
+                        completedStatus,
+                        restartSequenceNumber);
+
+                default:
+                    throw new NotSupportedException($"{testOperationResultType} is not supported in CreateEdgeHubRestartTestResult()");
+            }
+        }
+
+        static async Task<Tuple<DateTime, HttpStatusCode>> SendDirectMethodAsync(
+            string deviceId,
+            string targetModuleId,
+            ModuleClient moduleClient,
+            string directMethodName,
+            DateTime testExpirationTime,
+            CancellationToken cancellationToken)
+        {
+            while ((!cancellationToken.IsCancellationRequested) && (DateTime.UtcNow < testExpirationTime))
+            {
+                // BEARWASHERE -- TODO: Test this
+                try
+                {
+                    // Direct Method sequence number is always increasing regardless of sending result.
+                    Interlocked.Increment(ref directMethodCount);
```

Comment:
you don't need this lock; as this variable will not be accessed from multiple threads.

Reply:
It does from `SendDirectMethodAsync()`